datasets/tne/tne.py

```diff
@@ -0,0 +1,227 @@
+                            ),
+                        }
+                    ],
+                    "metadata": {
+                        "annotators": {
+                            "coref_worker": datasets.Value("int32"),
+                            "consolidator_worker": datasets.Value("int32"),
+                            "np-relations_worker": datasets.Sequence(datasets.Value("int32")),
+                        },
+                        "url": datasets.Value("string"),
+                        "source": datasets.Value("string"),
+                    },
+                }
+            )
+        return datasets.DatasetInfo(
+            # This is the description that will appear on the datasets page.
+            description=_DESCRIPTION,
+            # This defines the different columns of the dataset and their types
+            features=features,  # Here we define them above because they are different between the two configurations
+            # If there's a common (input, target) tuple from the features, uncomment supervised_keys line below and
+            # specify them. They'll be used if as_supervised=True in builder.as_dataset.
+            # supervised_keys=("sentence", "label"),
+            # Homepage of the dataset for documentation
+            homepage=_HOMEPAGE,
+            # License for the dataset if available
+            license=_LICENSE,
+            # Citation for the dataset
+            citation=_CITATION,
+        )
+
+    def _split_generators(self, dl_manager):
+        urls = _URLS
+        data_dir = dl_manager.download_and_extract(urls)
+
+        if self.config.name == "labeled":
+            return [
+                datasets.SplitGenerator(
+                    name=datasets.Split.TRAIN,
+                    # These kwargs will be passed to _generate_examples
+                    gen_kwargs={
+                        "filepath": data_dir["train"],
+                        "split": "train",
+                    },
+                ),
+                datasets.SplitGenerator(
+                    name=datasets.Split.VALIDATION,
+                    # These kwargs will be passed to _generate_examples
+                    gen_kwargs={
+                        "filepath": data_dir["dev"],
+                        "split": "dev",
+                    },
+                ),
+            ]
+        else:
+            return [
+                datasets.SplitGenerator(
+                    name=datasets.Split.TEST,
+                    # These kwargs will be passed to _generate_examples
+                    gen_kwargs={"filepath": data_dir["test_unlabeled"], "split": "test_unlabeled"},
+                ),
+                datasets.SplitGenerator(
+                    name=datasets.Split("test_ood"),
+                    # These kwargs will be passed to _generate_examples
+                    gen_kwargs={"filepath": data_dir["ood_unlabeled"], "split": "ood_unlabeled"},
+                ),
+            ]
```

Comment:
Do you think it could make sense to have only one single configuration with the four splits in it ? It would seem more natural to me.

You can set `nps` to None or empty list when there's no label

Reply:
I figured it out. I just had to change the `_generate_examples` function to set an empty list in the case of the test files.